### ISSUE TITLE
Some subfolders will never sync

### ISSUE THREAD

**Issue by GAS85** [icewind1991/files_external_dropbox#18]
Hi, I noticed that after latest master upgrade, root sync issue was solved, but I've noticed that API2 Dropbox folder is smaller than it should be...
![image](https://user-images.githubusercontent.com/6813635/35222096-f615e97c-ff7c-11e7-88d3-25550ef380c4.png)

The thing is - I still have APIv1 enabled and use both of them, but it is time limited:
![image](https://user-images.githubusercontent.com/6813635/35222160-32cb37dc-ff7d-11e7-8455-e2a3c704f874.png)

If I discover subfolders, some of them has different size. I could not really find what exactly is wrong with not synced folders, like path is too long or files fas some strange symbols in names, nothing.
![image](https://user-images.githubusercontent.com/6813635/35222577-aeedb42e-ff7e-11e7-9acc-e35589d62bcf.png)

`php occ files:scan --all` was executed many times, does not solve the issue.

**Comment by icewind1991** [comment]
Can you provide the names of some of the incorrect folders?

**Comment by GAS85** [comment]
I found some interesting thing, all corrupted folder are listed on APIv2 with creation date 01.01.1970:
This I can see only if connect via webdav
```
... webdav/Dropbox-gas_A2/Other/OLD$ ls -la
итого 1080
drwx------ 1 user user      0 Dez 29 11:38 .
drwx------ 1 user user      0 Dez 29 11:38 ..
...
drwx------ 1 user user      0 Jan  1  1970 Hurgada
...
drwx------ 1 user user      0 Jan  1  1970 Nsk 22122013
```
So path will be like 
.../Dropbox-gas_A2/Other/OLD/Hurgada

Or this is other examples:
```
webdav/Dropbox-gas_A2/Other/OLD$ ls -la ../../Share/ | grep 1970
drwx------ 1 user user        0 Jan  1  1970 Knaus

```
path is /Dropbox-gas_A2/Share/Knaus
and
/Dropbox-gas_A2/Share/Master
Nothing special - 0 Files.

I will try to recreate APIv2 Dropbox. Could it be that issue comes because of update of this app?

**Comment by realslacker** [comment]
I seem to be seeing this as well. In my case it's files in my [YNAB](http://www.youneedabudget.com) folder.

```
Dropbox:\Apps\... (these are ok)
Dropbox:\Camera Uploads\... (these are ok)
Dropbox:\YNAB\Exports\... (files missing, directory empty)
Dropbox:\YNAB\Lisa's Budget 2018~5F8411EF.ynab4\... (files and direcotries missing, directory empty)
Dropbox:\YNAB\My Budget 2018~9D9A9073.ynab4\... (files and direcotries missing, directory empty)
Dropbox:\YNAB\My Budget~C022A36D.ynab4\... (files and direcotries missing, directory empty)
```

**Comment by GAS85** [comment]
Just checked with other user that we have Dropbox shares... The same folder is corrupted "/Share/Master". But for me it is empty, and for user 2 it contains subfolders, bot 0 files.
Originally it has subfolders and files.

**Comment by Flori1989** [comment]
It looks like all directories whose names begin with uppercase letters are not synced. Can anyone confirm that?

**Comment by GAS85** [comment]
Not really, I have more folders with uppercase letters at the begging, but not all of them have trouble.

**Comment by icewind1991** [comment]
Can you try re-scanning the storage.

```
occ files:scan --path /$user/files/Path/To/Dropbox
```

**Comment by GAS85** [comment]
I did it before and I have a cron job twice a day to run this rescan scrip: https://github.com/GAS85/nextcloud_scripts/blob/master/nextcloud-file-sync.sh there was no positive effect at all. I will try to run command manually this evening to do the test.

**Comment by hpfmn** [comment]
For me is also one folder not synced and this is a folder that is also synced with another app. It is all small letters - so it has nothing to do with capital letters.

**Comment by hpfmn** [comment]
If it is of any interest  I used the owncloud version of this plugin and didn't have the issue for me.

**Comment by GAS85** [comment]
Size is still 0 for same folders.

**Comment by zpv** [comment]
@GAS85 
I took the latest plugin from ownCloud and made it compatible with Nextcloud using icewind1991's original patch.

https://github.com/zpv/files_external_dropbox

It has been working fine for me so far and fixes the missing files/subdirectories the current version of this experiences.

**Comment by hpfmn** [comment]
@zpv it is not possible to activate your plugin... but if just updating the base helps maybe @icewind1991 should consider this?

**Comment by zpv** [comment]
@hpfmn Latest commit should be working now. Forgot to remove a line in the metadata regarding ownCloud dependency.

**Comment by hpfmn** [comment]
I think the problem is commit 03ba610c21197453800d8f805a1e29f55c98f4e8 - for some reason it helps to put everything to lower case.
